### ISSUE TITLE
Whole system crashes when scrolling

### ISSUE THREAD

**Issue by ClaudiaKre** [phase1geo/Minder#457]
Hello,

I use Ubuntu (18.04 LTS at my desktop computer and 20.04 LTS at my notebook). I installed minder 1.14.0 from flathub. I made a very big minder file (927 kb). The problem is, when I now open the file in minder, at the beginning it works well and I can mark different nodes and modify them. But when I begin scroll, the whole system crashes immediately. Neither the gnome shell nor the tty console works after the crash, so I'm also unable to kill the application or to get an idea what is the reason.

The very strange thing is. I first noticed that behavior at my desktop computer and I was no longer able to 'work' at that file, but it still worked on my notebook and so I only used it there. Now, the file growed and since yesterday my notebook show exactly the same problem.

Can anybody tell me, how to fix the problem? Can I fix the bug by myself?

Greetings,
Claudia

**Comment by dirk-w** [comment]
Hi,

I have the same Problem (maybe)
My system is Ubuntu 18.04 LTS and I have installed the newest version 14.0.0
After adding 1 subnode my whole system crashes. so not a real great mindmap.
What I see is, that it seems not to store any kind of data (I have tried to do a "sav as" select a folder in my home dir but nothing happend)
Strange that the whole system crashes. are there any kind of logfiles?

is it possible to fix this??

Thanks 
Cheers 
Dirk

**Comment by ClaudiaKre** [comment]
Hello,

I have a new insight. After my system (both, at desktop computer with Ubuntu 18.04 and notebook with Ubuntu 20.04) crashed the first time, the application doesn't work at all anymore. When I now open an older file (that worked one week ago without problems) it now crashes, too, when scrolling. Even after deinstalling/reinstalling of minder and after deleting all minder folders to 'reset' it ... minder does now not open anything without problems.

So as it seems to me minder did trigger any 'bigger' problem that still remains after resetting deinstalling ... on both of my systems.

Greetings,
Claudia

**Comment by phase1geo** [comment]
Out of curiosity, has anyone here tried to build Minder from source?  If you are using a Flatpak, it would be preferable to run the Flatpak with the elementary runtime I don't know that the one maintained on Flathub uses this runtime).

In addition, make sure that you have permissions to write and read your user folders via a portal.

**Comment by ClaudiaKre** [comment]
Hi,
I'm only a 'user' without many insights into computing. I gave the building a short try, but I didn't know who to install the dependencies (my system didn't find 'libhandy-1-dev' for example). But what exactly do you mean with "elementary runtime"? I thought if there has to be a special runtime for minder, it should be part of the dependecies or in this case, of the flatpak, or not?

I didn't hear of 'portal' before and I will read up about it today and then tell you more. But what I can say so far: I could save the files, as well as load the files. Loading by doubleclicking the file was a bit buggy, in many cases, I saw map flashing up for a millisecond and then I found a new empty map. But the "open" dialog did always work (and still works now, but I'm not allowed to scroll).

**Comment by ClaudiaKre** [comment]
Hello again,

yesterday I tried to figure out, if the crash problem is Ubuntu-specific:
I installed Elementary OS 6.1 in a virtual machine (gnome-boxen) and I installed Minder and opened my problematic files. And I had the exactly same outcome: The whole system crashed reliably. I tested it now for five times with slightery different versions of the file, but it crashed everytime - in the most cases in the first minute.

So I think it's nothing about Ubuntu but about my files. I saved them in a 'normal' minder-typical way and did no modifications to it.

@phase1geo do you have an idea what the problem is or how I can fix the files?

[Minder-files not working.zip](https://github.com/phase1geo/Minder/files/8040448/Minder-files.not.working.zip)

**Comment by phase1geo** [comment]
Okay, so I loaded the first file (Wirbeltiere_v71_Familien....minder).  I've been scrolling around for about five minutes, folding/unfolding, moving into and out of focus mode, zooming in/out.  So far, the application has been running (no crashes) for me.  This is running on a VM of elementary OS 6.1 (latest master branch) which is basically 1.14.0 with a couple of feature improvements (no bug fixes since 1.14.0).  I am running the application from source (so that I can debug the application if segfaults -- debugging Flatpaks is difficult in comparison).

**Comment by ClaudiaKre** [comment]
hey @phase1geo ,
I'm sorry, that you spent so much time. Thank you for that. That you had no crashes is magic for me ... same system, same Minder version. The only thing that I can guess is, that it is connected to the flatpak. Did you use the first file, of three in the zip-file that I sent in my last post? Would you have a last look on the other two files (one minute should absolute be enough!). I remember, that the crash for the first (the oldest) file took a longer time on my virtual system to crash then the second. And the third file (the newest) I was unable to open at all (there it crashed during the opening). If you see no crashes there, then I'm happy, that the files are not corrupted and can somehow get rescued (I worked many weeks for that) but then I have no idea what to do.

Du you have a compiled (non flatpak) version of minder, that I could install in the virtual machine?

**Comment by dirk-w** [comment]
Hi and thanks a lot for support and programming this tool,

as I understand the problem is the flatpak and the best way is to compile the sources, right?
That can be done by
cloning the repo (git clone....)
./app install-deps
./app install
./app run

I try to remove the flatpak and install minder this weekend
What do I have to do when new version is released? same procedure? 
git fetch... 
./app install (or maybe update the dependency's? 

Thanks 
Cheers
Dirk

**Comment by phase1geo** [comment]
@ClaudiaKre So I had some time this evening.  I ran the Flatpak with all three documents opened.  I played around with all of them for at least 15 minutes without any issues.  This is running on elementary OS on bare metal.

**Comment by phase1geo** [comment]
@dirk-w Yes, I think what you are describing is what you would need to do if you were running everything from GitHub.

**Comment by ClaudiaKre** [comment]
Bummer, that we can't solve the problem with the flatpaks. Meanwhile I compiled minder with elementary OS (in a virtual machine) and now it works!
Sad, that I can't use it directly with Ubuntu but I hope it's a bug in the flatpak runtime and someone will somehow fix it sometime ...
